changed the default value for allow-privileged for the kubelet (kuber…

CalvinHartwell opened the pull request:
**What this PR does / why we need it**:

This PR modifies the default value for the kubernetes-worker: it sets the allow-privileged kubelet value to true, based on this issue:[https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/579](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/579). 

The original PR was here: [https://github.com/kubernetes/kubernetes/pull/63442](https://github.com/kubernetes/kubernetes/pull/63442) which included this change for the k8s 1.10 release. This PR incorporates this fix into the Canonical distribution of Kubernetes. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

This change fixes two issues: 

[https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/579](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/579)
[https://github.com/rancher/rancher/issues/13612](https://github.com/rancher/rancher/issues/13612)

**Special notes for your reviewer**:

Waiting review from Canonical Kubernetes charm tea. 

**Release note**:

```release-note
The new default value for the --allow-privileged parameter of the Kubernetes-worker charm has been set to true based on changes which went into the Kubernetes 1.10 release. Before this change the default value was set to false. If you're installing Canonical Kubernetes you should expect this value to now be true by default and you should now look to use PSP (pod security policies). 
```

CalvinHartwell committed:
changed the default value for allow-privileged for the kubelet (kubernetes-worker) based on new standard for 1.10 release

hyperbolic2346 commented on cluster/juju/layers/kubernetes-worker/config.yaml:
```diff
@@ -13,7 +13,7 @@ options:
-    default: "auto"
+    default: true
     description: |
       Allow privileged containers to run on worker nodes. Supported values are
```
We should update the description to deprecate this flag. I think we should also remove auto, but maybe that's another issue.

CalvinHartwell replied:
@hyperbolic2346 yes - I can update the description if you wish, and I also think the auto option should be removed, but perhaps this can be another PR. 

Will you approve if I make those changes?

Cynerva replied:
I've opened https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/585 to address the deprecation of `allow-privileged`. We can follow up on that later. :+1:

Cynerva commented:
/ok-to-test
/assign @Cynerva

Cynerva commented:
`default: true` could be problematic. Because in yaml, `true` is a boolean, not a string (even though you can normally omit quotes and get a string. don't get me started). It would be safer to have this as `default: "true"`

@CalvinHartwell could you add the quotes back in? It's possible that Juju and the kubernetes-worker charm are able to handle this gracefully, but I'd rather not rely on that.

CalvinHartwell committed:
Merge pull request #1 from kubernetes/master

Adding recent upstream changes to k8s.

CalvinHartwell commented:
@Cynerva @hyperbolic2346 added the quotes back in, ready for testing and merge if ALL ok!

Cynerva commented:
@CalvinHartwell Where did that `cluster-context` config come from? That wasn't in this PR before.

CalvinHartwell committed:
fixed branch and changed values to true

CalvinHartwell commented:
@Cynerva apologies, accidentally a merged an old branch into the mix. I've now fixed that and appended some context to the readme section. More importantly I've also adjusted the same value for the kubernetes-master as this is also causing an issue for workloads. 

Looking good?

CalvinHartwell committed:
fixed the bad branch merge issue

Cynerva commented on cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py:
```diff
@@ -1591,4 +1591,4 @@ def _write_gcp_snap_config(component):
         daemon_env += '{}={}\n'.format(gcp_creds_env_key, creds_path)
         daemon_env_path.parent.mkdir(parents=True, exist_ok=True)
-        daemon_env_path.write_text(daemon_env)
+        daemon_env_path.write_text(daemon_env)
```
I'd let this slide, but our automation runs flake8 when building the charms, and the newline change here will cause it to fail:

```
$ flake8 reactive/
reactive/kubernetes_master.py:1594:47: W292 no newline at end of file
```

CalvinHartwell committed:
fixed newline issue

CalvinHartwell committed:
fixed newline issue

Cynerva commented:
/lgtm

Thanks!

CalvinHartwell commented:
@thockin @eparis any ideas on whats wrong with the E2E tests?
